### PR TITLE
Gradle 2.9 unsupported error

### LINKED ISSUE #581: Warning that Gradle 2.9 is unsupported
Until #568 is fixed, the plugin should warn explaining the reason for the failure.

### PR DESCRIPTION
**Pull request by brunobowden**
- Throws explanatory error for gradle version 2.9 or later
- Meant to be temporary until #568 is fixed
- Fixes #581

### CODE REVIEW

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/NativeCompilation.groovy [hunk 1]
```diff
@@ -121,6 +122,18 @@ class NativeCompilation {
         })
     }
 
+    // Creates fake tasks to satisfy dependencies for J2objcPlugin.apply(...)
+    // Should only be used as an alternative to NativeComplication.apply(...) when
+    // the Gradle Version is unsupported and no build can be done.
+    void applyForUnsupported() {
```
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/NativeCompilation.groovy]: applyWhenUnsupported I think reads better.

**Comment by advayDev1** [comment]
LGTM with comment

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/NativeCompilation.groovy (hunk 1 above)
Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/NativeCompilation.groovy]: Duly noted and modified

**Comment by brunobowden** [comment]
I was surprised to see that the checks introduced a deadlock. I'm glad this is sorted now. I'll merge once the CI passes.

**Pull request merged**